src/org/gk/pro/ProExporter.java

```diff
@@ -0,0 +1,643 @@
+        location = "GO:0005654";
+        accession = "Q9UIF7-6";
+        start = 1;
+        end = 521;
+        modifications = "+255=MOD:01643+255=MOD:00029";
+        freeText = "";
+        displayName = "MUTYH-6 M255V [nucleoplasm]";
+        output = getTestRow(type, identifier, location, accession, start, end, modifications, freeText, displayName);
+        modifiedResidues = ewas.getAttributeValuesList(ReactomeJavaConstants.hasModifiedResidue);
+        assertEquals(output, getRow(ewas, modifiedResidues));
+
+        // IntraChainCrosslinkedResidue #1
+        ewas = dba.fetchInstance(8874904L);
+        identifier = "R-HSA-8874904";
+        location = "GO:0005758";
+        accession = "Q9NRP2";
+        start = 1;
+        end = 79;
+        modifications = "+14=MOD:00798[CROSSLINK1@47]+24=MOD:00798[CROSSLINK2@37]+14=CHEBI:23514[CROSSLINK1@47]+24=CHEBI:23514[CROSSLINK2@37]";
+        freeText = "CROSSLINK1=Intra-chain Crosslink via half cystine at 14 and 47^|^CROSSLINK2=Intra-chain Crosslink via half cystine at 24 and 37";
+        displayName = "4xHC-CMC2 [mitochondrial intermembrane space]";
+        output = getTestRow(type, identifier, location, accession, start, end, modifications, freeText, displayName);
+        modifiedResidues = ewas.getAttributeValuesList(ReactomeJavaConstants.hasModifiedResidue);
+        assertEquals(output, getRow(ewas, modifiedResidues));
+
+        // IntraChainCrosslinkedResidue #2
+        ewas = dba.fetchInstance(6797422L);
+        identifier = "R-HSA-6797422";
+        location = "GO:0005576";
+        accession = "P09429";
+        start = 2;
+        end = 215;
+        modifications = "+23=MOD:00798[CROSSLINK1@45]+23=CHEBI:30770[CROSSLINK1@45]";
+        freeText = "CROSSLINK1=Intra-chain Crosslink via half cystine at 23 and 45";
+        displayName = "HC23,45-HMGB1 [extracellular region]";
+        output = getTestRow(type, identifier, location, accession, start, end, modifications, freeText, displayName);
+        modifiedResidues = ewas.getAttributeValuesList(ReactomeJavaConstants.hasModifiedResidue);
+        assertEquals(output, getRow(ewas, modifiedResidues));
+
+        // IntraChainCrosslinkedResidue #3
+        ewas = dba.fetchInstance(8874912L);
+        identifier = "R-HSA-8874912";
+        location = "GO:0005758";
+        accession = "Q9BSY4";
+        start = 1;
+        end = 110;
+        modifications = "+58=MOD:00798[CROSSLINK1@89]+58=CHEBI:23514[CROSSLINK1@89]";
+        freeText = "CROSSLINK1=Intra-chain Crosslink via half cystine at 58 and 89";
+        displayName = "4xHC-CHCHD5 [mitochondrial intermembrane space]";
+        output = getTestRow(type, identifier, location, accession, start, end, modifications, freeText, displayName);
+        modifiedResidues = ewas.getAttributeValuesList(ReactomeJavaConstants.hasModifiedResidue);
+        assertEquals(output, getRow(ewas, modifiedResidues));
+
+    }
+
+    @Test
+    public void testGetFreeText() throws Exception {
```

Comment:
Could you move all these long test methods into another class?

Reply:
All tests have been moved to `org.gk.pro.ProExporterTest`